app/src/main/java/org/surrel/facebooknotifications/PrefsActivity.java

```diff
@@ -1,28 +1,64 @@
 package org.surrel.facebooknotifications;
 
 import android.content.SharedPreferences;
+import android.content.SharedPreferences.Editor;
 import android.os.Bundle;
+import android.preference.EditTextPreference;
 import android.preference.PreferenceActivity;
 import android.preference.PreferenceManager;
 import android.util.Log;
+import android.widget.Toast;
 
 public class PrefsActivity extends PreferenceActivity implements SharedPreferences.OnSharedPreferenceChangeListener {
+    public static final String ENABLE_NOTIFICATION_SYNCHRO = "enable_notification_synchro";
+    public static final String UPDATE_INTERVAL = "update_interval";
+    public static final int DEFAULT_UPDATE_INTERVAL = 15;
+    public static final int MIN_UPDATE_INTERVAL = 1;
+    public static final int MAX_UPDATE_INTERVAL = 10080;
+    private EditTextPreference updatePreference;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         //It is deprecated but it will work, it is still only way for lower android versions
         addPreferencesFromResource(R.xml.preferences);
         PreferenceManager.getDefaultSharedPreferences(this).registerOnSharedPreferenceChangeListener(this);
+        updatePreference = (EditTextPreference) findPreference(UPDATE_INTERVAL);
     }
 
     @Override
     public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
 
         Log.d("fbn.PrefsActivity", key);
 
-        if (key.equals("enable_notification_synchro")
-                || key.equals("update_interval")) {
+        if (key.equals(ENABLE_NOTIFICATION_SYNCHRO)) {
+            Log.d("fbn.PrefsActivity", "Calling WakeupManager");
+            WakeupManager.updateNotificationSystem(this);
```

Comment:
I don't understand. I cant see these duplicated. Aren't these lines from an intermediary commit before i switched to a switch instead of an if cascade?